test/unit/logic/vote.js

```diff
@@ -522,6 +522,94 @@ describe('vote', () => {
 		});
 	});
 
+	describe('getBytes', () => {
+		it('should throw error for wrong votes asset format', () => {
+			var transaction = _.cloneDeep(validTransaction);
+			transaction.asset = {};
+			return expect(() => {
+				vote.getBytes(transaction);
+			}).to.throw('votes parameter must be an Array');
+		});
+
+		it('should throw error for zero votes', () => {
+			var transaction = _.cloneDeep(validTransaction);
+			transaction.asset.votes = [];
+			return expect(() => {
+				vote.getBytes(transaction);
+			}).to.throw('number of votes must be between 0 and 33.');
+		});
+
+		it('should throw error for votes greater than 33', () => {
+			var transaction = _.cloneDeep(validTransaction);
+			const votes = new Array(34);
+			_.fill(
+				votes,
+				'-9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f'
+			);
+			transaction.asset.votes = votes;
+			return expect(() => {
+				vote.getBytes(transaction);
+			}).to.throw('number of votes must be between 0 and 33.');
+		});
+
+		it('should throw error for votes with invalid public key', () => {
+			var transaction = _.cloneDeep(validTransaction);
+			transaction.asset.votes = ['-9d3058175acab969f41ad9b86f7a2926c74258'];
+			const publicKey = transaction.asset.votes[0].substr(
+				1,
+				transaction.asset.votes[0].length
+			);
+			const err = `Public key ${publicKey} length:${
+				publicKey.length
+			} differs from the expected 64 characters.`;
+
+			return expect(() => {
+				vote.getBytes(transaction);
+			}).to.throw(err);
+		});
+
+		it('should throw error for missing prefix +/-', () => {
+			var transaction = _.cloneDeep(validTransaction);
+			transaction.asset.votes = [
+				'$9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f',
+			];
+			const prefix = transaction.asset.votes[0].charAt(0);
+			const err = `Expected a prefix '+' for vote and '-' for unvote in public key, but found ${prefix}`;
+
+			return expect(() => {
+				vote.getBytes(transaction);
+			}).to.throw(err);
```

Comment:
Can you update the other tests to have the same pattern, the error message in a declared string and point to it in the assertion?

Reply:
the getBytes is not used in the `logic/votes` functions so I don't see any other place where error being thrown, or am I missing something?

Reply:
Your approach works fine :+1: